simplex/bls.go

```diff
@@ -0,0 +1,119 @@
+package simplex
+
+import (
+	"encoding/binary"
+	"fmt"
+	"simplex"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+)
+
+var (
+	simplexLabel = []byte("<simplex>")
+)
+
+// BLSSigner signs messages encoded with the provided ChainID and NetworkID
+// using the SignBLS function.
+type BLSSigner struct {
+	chainID  ids.ID
+	subnetID ids.ID
+	signBLS  func(msg []byte) (*bls.Signature, error) // should we pass in a verifier function into BLSVerifier?
```

Comment:
the signer function here is because we support both software signing and hardware signing. When we verify, we only verify with software, so no function is needed